eventmetrics/parsers/provide.go

```diff
@@ -256,46 +336,104 @@ func createBootDurationCallback(m Measures) (func(interpreter.Event, float64), e
 	}, nil
 }
 
-func createRebootToManageableCallback(m Measures) (func(interpreter.Event, interpreter.Event, float64), error) {
-	if m.RebootToManageableHistogram == nil {
-		return nil, errNilRebootHistogram
+func createTimeElapsedCallback(m Measures, name string) (func(interpreter.Event, interpreter.Event, float64), error) {
+	if m.TimeElapsedHistograms == nil {
+		return nil, errNilHistogram
+	}
+
+	if _, found := m.TimeElapsedHistograms[name]; !found {
+		return nil, errNilHistogram
 	}
 
 	return func(currentEvent interpreter.Event, startingEvent interpreter.Event, duration float64) {
 		labels := getTimeElapsedHistogramLabels(currentEvent)
-		m.RebootToManageableHistogram.With(labels).Observe(duration)
+		histogram := m.TimeElapsedHistograms[name]
+		histogram.With(labels).Observe(duration)
 	}, nil
 }
 
-func createEventValidator(config RebootParserConfig) validation.Validator {
-	config = checkTimeValidations(config)
+func createEventValidator(config EventValidationConfig) (validation.Validator, error) {
+	validationType := enums.ParseValidationType(config.Key)
+	if validationType == enums.UnknownValidation {
+		return nil, errNonExistentKey
+	}
+
+	switch validationType {
+	case enums.BootTimeValidation:
+		config = checkTimeValidations(config)
+		bootTimeValidator := validation.TimeValidator{
+			Current:      time.Now,
+			ValidFrom:    config.BootTimeValidator.ValidFrom,
+			ValidTo:      config.BootTimeValidator.ValidTo,
+			MinValidYear: config.BootTimeValidator.MinValidYear,
+		}
+		return validation.BootTimeValidator(bootTimeValidator), nil
+	case enums.BirthdateValidation:
+		config = checkTimeValidations(config)
+		birthdateValidator := validation.TimeValidator{
+			Current:      time.Now,
+			ValidFrom:    config.BootTimeValidator.ValidFrom,
+			ValidTo:      config.BootTimeValidator.ValidTo,
+			MinValidYear: config.BootTimeValidator.MinValidYear,
+		}
+		return validation.BirthdateValidator(birthdateValidator), nil
+	case enums.MinBootDuration:
+		config = checkTimeValidations(config)
+		return validation.BootDurationValidator(config.MinBootDuration), nil
+	case enums.BirthdateAlignment:
+		config = checkTimeValidations(config)
+		return validation.BirthdateAlignmentValidator(config.BirthdateAlignmentDuration), nil
+	case enums.ValidEventType:
+		return validation.EventTypeValidator(config.ValidEventTypes), nil
+	case enums.ConsistentDeviceID:
+		return validation.ConsistentDeviceIDValidator(), nil
+	default:
+		return nil, errWrongEventValidatorKey
+	}
+}
 
-	bootTimeValidator := validation.TimeValidator{
-		Current:      time.Now,
-		ValidFrom:    config.BootTimeValidator.ValidFrom,
-		ValidTo:      config.BootTimeValidator.ValidTo,
-		MinValidYear: config.BootTimeValidator.MinValidYear,
+func createCycleValidator(config CycleValidationConfig) (history.CycleValidator, error) {
+	validationType := enums.ParseValidationType(config.Key)
```

Comment:
are the validation type enums used for anything other than this? It seems like cycle validator types are completely separate from event validators and should have their own enum?

Reply:
No, the validation type enums are just used for the validator creation from the config. Having separate cycle validator type sounds good!